phoenix-core/src/main/java/org/apache/phoenix/mapreduce/OrphanViewTool.java

```diff
@@ -0,0 +1,812 @@
+    }
+
+    /**
+     * Go through all the views in the system catalog table and add them to orphanViewSet
+     * @param phoenixConnection
+     * @throws Exception
+     */
+    private void populateOrphanViewSet(PhoenixConnection phoenixConnection)
+            throws Exception {
+        ResultSet viewRS = phoenixConnection.createStatement().executeQuery(viewQuery);
+        while (viewRS.next()) {
+            String tenantId = viewRS.getString(1);
+            String schemaName = viewRS.getString(2);
+            String tableName = viewRS.getString(3);
+            Key key = new Key(tenantId, schemaName, tableName);
+            View view = new View(key);
+            orphanViewSet.put(key, view);
+        }
+    }
+
+    /**
+     * Go through all the tables in the system catalog table and update baseSet
+     * @param phoenixConnection
+     * @throws Exception
+     */
+    private void populateBaseSet(PhoenixConnection phoenixConnection)
+            throws Exception {
+        ResultSet baseTableRS = phoenixConnection.createStatement().executeQuery(candidateBaseTableQuery);
+        while (baseTableRS.next()) {
+            String tenantId = baseTableRS.getString(1);
+            String schemaName = baseTableRS.getString(2);
+            String tableName = baseTableRS.getString(3);
+            Key key = new Key(tenantId, schemaName, tableName);
+            Base base = new Base(key);
+            baseSet.put(key, base);
+        }
+    }
+
+    /**
+     * Go through all the physical links in the system catalog table and update the base table info of the
+     * view objects in orphanViewSet. If the base or view object does not exist for a given link, then add the link
+     * to orphanLinkSet
+     * @param phoenixConnection
+     * @throws Exception
+     */
+    private void processPhysicalLinks(PhoenixConnection phoenixConnection)
+            throws Exception {
+        ResultSet physicalLinkRS = phoenixConnection.createStatement().executeQuery(physicalLinkQuery);
+        while (physicalLinkRS.next()) {
+            String tenantId = physicalLinkRS.getString(1);
+            String schemaName = physicalLinkRS.getString(2);
+            String tableName = physicalLinkRS.getString(3);
+            Key viewKey = new Key(tenantId, schemaName, tableName);
+            View view = orphanViewSet.get(viewKey);
+
+            String baseTenantId = physicalLinkRS.getString(4);
+            String baseFullTableName = physicalLinkRS.getString(5);
+            Key baseKey = new Key(baseTenantId, baseFullTableName);
+            Base base = baseSet.get(baseKey);
+
+            if (view == null || base == null) {
+                orphanLinkSet.add(new Link(viewKey, baseKey, PTable.LinkType.PHYSICAL_TABLE));
+            }
+            else {
+                view.setBase(baseKey);
+            }
+        }
+    }
+
+    /**
+     * Go through all the child-parent links in the system child-link table and update the parent field of the
```

Comment:
Nit: This comment should say "system catalog table" instead of "child-link", right?

Reply:
Yes